### PR TITLE
docs: update missing help sections from main help page

### PR DESCRIPTION
**Pull request by dundargoc**
Also replace non-existent help section remote.txt to remote_plugins.txt

**NOTE:** I added the `OTHER` section as a placerholder until someone comes up with a better idea on how to categorize each help section.

**NOTE 2:** I skipped the help texts `pi_tutor.txt`, `lsp-extension.txt`, `if_lua.txt` and `msgpack_rpc.txt` since I didn't feel their quality was up to par. Tell me if you disagree and I'll add them.

### CODE REVIEW

File: runtime/doc/help.txt [hunk 1]
```diff
@@ -164,13 +167,24 @@ GUI ~
...
 |if_cscop.txt|	using Cscope with Vim
+|if_perl|	Perl interface
 |if_pyth.txt|	Python interface
 |if_ruby.txt|	Ruby interface
 |sign.txt|	debugging signs
 
 Versions ~
 |vim_diff.txt|	Main differences between Nvim and Vim
 |vi_diff.txt|	Main differences between Vim and Vi
+|deprecated.txt|  Deprecated items that has been or will be removed
```
Comment (gpanders) [inline runtime/doc/help.txt:178]:
```suggestion
|deprecated.txt|  Deprecated items that have been or will be removed
```

**Review by gpanders — APPROVED** [review]

**Review by clason — APPROVED** [review]

File: runtime/doc/help.txt [hunk 2]
```diff
@@ -164,13 +167,24 @@ GUI ~
 
 Interfaces ~
 |if_cscop.txt|	using Cscope with Vim
+|if_perl|	Perl interface
```
Comment (gpanders) [inline runtime/doc/help.txt:170]: The others in this list contain the `.txt` extension, should this one too?

Comment (dundargoc) [reply runtime/doc/help.txt]: Yep. Thanks.

**Pull request merged**